code/datums/materials/pizza.dm

```diff
@@ -0,0 +1,31 @@
+/datum/material/pizza
+	name = "pizza"
+	id = "pizza"
+	desc = "~Jamme, jamme, n'coppa, jamme ja! Jamme, jamme, n'coppa jamme ja, funi-culi funi-cala funi-culi funi-cala!! Jamme jamme ja funiculi funicula!~"
+	color = "#FF9F23"
+	categories = list(MAT_CATEGORY_RIGID = TRUE)
+	sheet_type = /obj/item/stack/sheet/pizza
+	value_per_unit = 0.05
+	beauty_modifier = 0.1
+	strength_modifier = 0.7
+	armor_modifiers = list("melee" = 0.3, "bullet" = 0.3, "laser" = 1.2, "energy" = 1.2, "bomb" = 0.3, "bio" = 0, "rad" = 0.7, "fire" = 1, "acid" = 1)
+	item_sound_override = 'sound/effects/meatslap.ogg'
+	turf_sound_override = FOOTSTEP_MEAT
+	texture_layer_icon_state = "pizza"
+
+/datum/material/pizza/on_removed(atom/source, material_flags)
+	. = ..()
+	qdel(source.GetComponent(/datum/component/edible))
```

Comment:
`GetComponent` bad. Send a signal at the base of `on_removed` and register to it on the component.

Reply:
GetComponent is actually fine in cases like this as it's essentialy the same as addcomponent. Binding this to a signal is not really a solution viable here as you might not be removing all materials.